### PR TITLE
Add resize callback to Screen

### LINKED ISSUE #182: Add resize callback to Screen
It seems to me that `Screen` needs a resize callback. Right now the only way I see to handle that event is to subclass `Screen`, which seems like an overkill for such a simple operation. (To be clear, I'm volunteering to do this; I'm starting this issue to make sure I'm not missing anything.)

### PR DESCRIPTION
**Pull request by mrzv**
Fixes #182. The only missing bit are the changes to `python/py_doc.h` (and corresponding `D(...)` bit in `python/widgets.cpp`) because I can't get `doc/mkdoc_rst.py` to work.

---
This change is [Reviewable](https://reviewable.io/reviews/wjakob/nanogui/183)

### CODE REVIEW

File: include/nanogui/screen.h [hunk 1]
```diff
@@ -123,7 +123,11 @@ class NANOGUI_EXPORT Screen : public Widget {
     virtual bool keyboardCharacterEvent(unsigned int codepoint);
 
     /// Window resize event handler
-    virtual bool resizeEvent(const Vector2i &) { return false; }
+    virtual bool resizeEvent(const Vector2i& size) { if (mResizeCallback) mResizeCallback(size); return false; }
```
Comment (wjakob) [inline include/nanogui/screen.h]: One minor request: can you move the implementation of the virtual function to the ``.cpp`` file?

Comment (mrzv) [inline include/nanogui/screen.h]: Done, I think.

File: src/screen.cpp [hunk 2]
```diff
@@ -456,6 +456,12 @@ bool Screen::keyboardCharacterEvent(unsigned int codepoint) {
     return false;
 }
 
+bool Screen::resizeEvent(const Vector2i& size) {
+    if (mResizeCallback)
+        mResizeCallback(size);
```
Comment (wjakob) [inline src/screen.cpp]: Sorry, one more thing: return ``true`` when there is a callback? (as the event was handled)

Comment (mrzv) [inline src/screen.cpp]: Yup. Done.

**Pull request merged**